app/router.ts

```diff
@@ -5,10 +5,25 @@ import config from 'ember-get-config';
 
 const Router = EmberRouter.extend({
     metrics: service('metrics'),
+    currentUser: service('currentUser'),
+    features: service('features'),
 
     location: config.locationType,
     rootURL: config.rootURL,
 
+    willTransition(oldInfo, newInfo, transition) {
+        const to = transition.targetName;
+        const flag = `ember_${to}_page`;
+        if (flag in config.featureFlags) {
+            this.get('currentUser._setWaffle').perform().then(() => {
```

Comment:
An issue with this approach is the `.then()` callback will run sometime later -- ideally, when waffle is already loaded we hit the `reload()` synchronously, before `willTransition` returns. Maybe something like
```ts
if (currentUser.get('waffleLoaded')) {
  // check flag, reload()
} else {
  currentUser.doTheWaffle().then(() => {
    // check flag, reload()
  });
}
```
or you could wrap that in a method on current-user:
```ts
// current-user.ts
waffleCheck(flag: string, waffleBack: (enabled: boolean) => void) {
  if (this.get('waffleLoaded')) {
    waffleBack(this.get('features').isEnabled(flag));
  } else {
    this.get('_setWaffle').perform().then(() => {
      waffleBack(this.get('features').isEnabled(flag));
    });
  }
}

// router.ts
currentUser.waffleCheck(flag, enabled => {
  if (!enabled) {
    window.location.reload();
  }
});
```
It's also nice that that keeps the task internal. I'm not a fan of exposing tasks if we can avoid it; better to leave them implementation details.